### ISSUE TITLE
Sync docs based on folder structure

### ISSUE THREAD

**Issue by colmdoyle** [readmeio/rdme#663]
I might be alone in this, but I would really love to have the CLI infer the category > page > child page relationship based on the structure of the docs folder I upload. As an example, if I ran `docs . --version=x` in the parent directory here, it would

<img width="320" alt="Screenshot 2022-11-07 at 12 06 39" src="https://user-images.githubusercontent.com/406463/200307377-555b1e64-814f-4611-bafe-b279b3eb3b39.png">

- Create a category called "capabilities"
- Create a doc for each `*.md` file in the `capabilities` directory and assign it to the category from step 1
- for any subfolders where the directory name matches the slug of a doc in the category from step 1, create a new doc for every `*.md` file and make it a child page of the doc where the slug matches the folder.

Basically I want to use the CLI to abstract away the Readme UI and having to go there to generate category IDs and doc IDs is getting in the way of that workflow.

**Comment by rlewis-grax** [comment]
Is there any expectation for this to be worked on or discussed further? This tool is not usable for use cases that involve the management of documentation via an external VCS without this feature. 

In my specific use case, documentation authors have no access to Readme and thus any dependency on using the API or UI to pull slugs/IDs to manage structure is fatal to the usability of this tool.

**Comment by kanadgupta** [comment]
Hi @rlewis-grax — we're in the early stages of working on some major improvements to ReadMe's docs editing workflow so it plays much nicer with Git. Don't have an estimated timeline since it's a pretty big undertaking but it is definitely top of mind for us!

**Comment by emma-campbell** [comment]
+1 for this option and commenting to stay updated.

We also have a nested output from tools like [Typedoc](https://typedoc.org/) and [Typedoc-Plugin-Markdown](https://github.com/tgreyuk/typedoc-plugin-markdown). The folder structure makes it much more digestible and easier to navigate.

**Comment by vegaro** [comment]
+1 for this, we are building our own scripts to sync categories and docs orders based on folder structure, but there are a lot of gotchas, like the fact that categories order cannot be changed from API

**Comment by evanbb** [comment]
This is how I intuitively expected the CLI to work.

@kanadgupta you mentioned you were planning some improvements here - are there any updates you can share from the last year and a half?

**Comment by runnabro** [comment]
Thanks for the feedback y'all! We've made a lot of progress :)

I'll reach out to everyone via email today so you can get in on an alpha.
